### DISCUSSION TITLE
How to collect NTP Status?

### DISCUSSION THREAD

**Discussion by jmg011** [NetApp/harvest#989] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Would like to see the NTP related status metrics

**Describe alternatives you've considered**
Custom Monitor

**Additional context**
N/A

**Comment by ruanruijuan** [comment]
ntp-server-status-get-iter.ntp-server-status-info is probably what you are looking for.

**Comment by jmg011** [comment]
Yes. Thank you!

**Comment by jmg011** [comment]
@ruanruijuan Hey, I could enable the metric collection on cmode for ntp servers, but I could not find the api for 7-mode. I queried zapi apis and grepped for both nis & server apis. Can you help find the api for nis 7-mode?

**Comment by ruanruijuan** [comment, marked as answer]
@jmg011 I contacted an engineer in the CoreSW team of ONTAP who hasn't touched 7-mode for years and here is his answer:
Assuming your have ntp running , not timed, they may be able to simply probe the ntp server via an ntpq command from a client machine on the network.
# ntpq -p alward-vsim1.sim.rtp.netapp.com
remote refid st t when poll reach delay offset jitter
==============================================================================
ntp02.rtp.eng.n .INIT. 16 u - 64 0 0.000 0.000 0.000

And I find a related KB article (you will need to sign in to see the entire page)
https://kb.netapp.com/Advice_and_Troubleshooting/Data_Storage_Software/ONTAP_OS/How_to_verify_NTP_operation_in_Data_ONTAP_8_(7-Mode_and_clustered_Data_ONTAP)
To print the status of a remote NTP server, use the following command.
In 7-Mode, the hostname/IP address is the interface used to manage the controller.
In clustered Data ONTAP, this is the ‘node-mgmt’ LIF for the node.
host# ntpq -pn name_or_IP

Example output:

[root@host]# ntpq -pn pepper

     remote           refid      st t when poll reach   delay   offset jitter
==============================================================================
 1.2.3.4        .INIT.          16 u    -  182    0    0.000    0.000 0.000
 10.64.82.114   10.57.102.104    3 u    2   64  377    0.477   11.179 0.297
*10.57.102.104  10.32.32.20      2 u    9   64  377  149.245   16.827 4.262

**Comment by jmg011** [comment]
@ruanruijuan Thank you for the help.
